assets/src/main/java/bisq/asset/coins/PickleProtocol.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.Coin;
+import bisq.asset.DefaultAddressValidator;
+
+public class PickleProtocol extends Coin {
+    public PickleProtocol() {
+
+        super("PickleProtocol", "PCK", new DefaultAddressValidator());
```

Comment:
Do not use deprecated validator

Reply:
should I be  using ?
package bisq.asset.coins;

import bisq.asset.Base58BitcoinAddressValidator;
import bisq.asset.Coin;
import bisq.asset.NetworkParametersAdapter;

public class PickleProtocol extends Coin {

    public PickleProtocol() {
        super("PickleProtocol", "PCK", new Base58BitcoinAddressValidator(new PickleProtocolParams()));
    }


    public static class PickleProtocolParams extends NetworkParametersAdapter {

        public PickleProtocolParams() {
            addressHeader = 54;
            p2shHeader = 119;
            acceptableAddressCodes = new int[]{addressHeader, p2shHeader};
        }
    }
}

Reply:
package bisq.asset.coins;

import bisq.asset.AbstractAssetWithDefaultValidatorTest;

import org.junit.Test;


public class PickleProtocolTest extends AbstractAssetWithDefaultValidatorTest {
    
    public PickleProtocolTest() {
    super(new PickleProtocol());
}
@ Test
    public void testValidAddresses() {
    assertValidAddress("NvkiGCx55W3cGPq3Q2kyQp4dncFypGH7DP");
    assertValidAddress("Nqbi28Hs4WwdqjdEEJWFPg3PryDFReuNgF");
    assertValidAddress("P7nkiTEHb7PWZVxi165tuBwjRhN5XsdYij");
}
@ Test
    public void testInvalidAddresses() {
    assertInvalidAddress("");
    assertInvalidAddress("Pqbi28Hs4WwdqjdEEJWFPg3PryDFReuNgF");
    assertInvalidAddress("N7nkiTEHb7PWZVxi165tuBwjRhN5XsdYij");
}
}

Reply:
I don't know what your coin is, so I am unable to advise you what implementation you should pick.
As to tests you should extend `AbstractAssetTest`.